runtime/context.go

```diff
@@ -99,6 +100,25 @@ func AnnotateIncomingContext(ctx context.Context, mux *ServeMux, req *http.Reque
 	return metadata.NewIncomingContext(ctx, md), nil
 }
 
+func isValidGRPCMetadataKey(key string) bool {
+	for i := 0; i < len(key); i++ {
+		r := key[i]
+		if !(r >= 'a' && r <= 'z') && !(r >= '0' && r <= '9') && r != '.' && r != '-' && r != '_' {
+			return false
+		}
+	}
+	return true
+}
+
+func isValidGRPCMetadataTextValue(textValue string) bool {
+	for i := 0; i < len(textValue); i++ {
+		if textValue[i] < 0x20 || textValue[i] > 0x7E {
```

Comment:
Ditto please add a reference here to https://github.com/grpc/grpc/blob/4b05dc88b724214d0c725c8e7442cbc7a61b1374/doc/PROTOCOL-HTTP2.md in a comment. The exact definition I see is
```
 %x20-%x7E
```
or
```
space and printable ASCII
```

Reply:
Done.